packages/react-charts/src/components/ChartUtils/chart-patterns.tsx

```diff
@@ -328,13 +328,9 @@ export const getDefaultPatternProps = ({
   const defaultColorScale = getDefaultColorScale(colorScale, themeColorScale);
   let defaultPatternScale = patternScale;
   let isPatternDefs = !patternScale && hasPatterns !== undefined;
-  let patternId;
+  const patternId = React.useMemo(() => getPatternId(), []);
```

Comment:
This change is ok, but not sure about renaming getDefaultPatternProps. 

Why does this function need to be a hook? It's not a public API, just used internally.

Reply:
It may be possible to move `React.useMemo` to a new function?

Just looking for a `uniqueId` to be called once, so we use the same ID for each component instance.

Reply:
Never mind the comment above. If creating a hook solves the issue, I'm ok with it.

Reply:
The function needs to be a hook as it uses a hook, and hooks can only be used in React components or by extension other hooks. The `useX` naming convention comes from the '[Rules of Hooks'](https://reactjs.org/docs/hooks-rules.html), which allows linters and other tooling to pick them up as hooks.